libsolidity/analysis/FunctionCallGraph.cpp

```diff
@@ -204,6 +204,26 @@ bool FunctionCallGraphBuilder::visit(MemberAccess const& _memberAccess)
 	return true;
 }
 
+bool FunctionCallGraphBuilder::visit(BinaryOperation const& _binaryOperation)
+{
+	if (_binaryOperation.annotation().userDefinedFunction.set())
+	{
+		solAssert(*_binaryOperation.annotation().userDefinedFunction);
+		functionReferenced(**_binaryOperation.annotation().userDefinedFunction, true /* called directly */);
+	}
+	return true;
+}
+
+bool FunctionCallGraphBuilder::visit(UnaryOperation const& _unaryOperation)
+{
+	if (_unaryOperation.annotation().userDefinedFunction.set())
+	{
+		solAssert(*_unaryOperation.annotation().userDefinedFunction);
```

Comment:
Are these kind of asserts really needed here?

Reply:
Unfortunately yes. `userDefinedFunction` is `SetOnce<FunctionDefinition const*> so it can be unset as well as set to `nullptr` and these mean different things.

One way to avoid this would be to use `std::reference_wrapper` so that it can't be null (it's never supposed to be null anyway). Alternatively I suggested that maybe comparing with `nullptr` would cover both cases but @wechman did not do that so I suspect it turned out it does not work that way.